decred/dcrd
# blockchain: move block validation rule tests into fullblocktests (3/3).

This PR adds the following block validation rule error tests to fullblocktests:
- `ErrInvalidEarlyStakeTx`
	- Create block that tries to vote before stake validation height.

- `ErrImmatureSpend`
	- Create block that spends immature stakebase from a vote in a ticket purchase.
  - Create block that spends immature change from one ticket purchase in another ticket purchase.
- `ErrScriptValidation`
	- Create block with an invalid stake transaction signature script.
	- Create block with an invalid regular transaction signature script.

- `ErrDiscordantTxTree`
	- Create block that tries to spend an input expected to be in a different tx tree than the one given in the TxIn outpoint.

- `ErrNoTax`
	- Create block with no dev subsudy for coinbase transaction.
	- Create block with an incorrect dev subsidy output amount.

- `ErrMissingTxOut`
	- Create block that tries to buy a ticket with the block's coinbase transaction.
	- Create block with a transaction that has a zero value output.
	- Create block that double spends a transaction input in the stake tree.
	- Create block that double spends a transaction input that has already been spent in the stake tree.
  - Create block with a ticket purchase using output from disapproved block.
  - Create block that a regular transaction using output disapproved block.

- `ErrTicketUnavailable`
	- Create block with a vote that spends an unavailable ticket.

- `ErrTxSStxOutSpend`
	- Create block that tries to spend an ticket purchase output as a regular transaction.

- `ErrRegTxSpendStakeOut`
	- Create block that double spends a stake transaction input in a regular transaction.

- `ErrSSRtxPayeesMismatch`
	- Create block that has a revocation with more payees than expected.

- `ErrSSRtxPayees`
	- Create block that has a revocation paying more than the original amount to the committed address.
	- Create block that has a revocation using a corrupted pay-to-address script.

- `ErrInvalidSSRtx`
	- Create block that has a revocation for a voted ticket.
